pkg/scheduler/scheduler.go

```diff
@@ -144,19 +185,24 @@ func (s *Scheduler) schedule(ctx context.Context) {
 		cq := snapshot.ClusterQueues[e.ClusterQueue]
 		if cq.Cohort != nil {
 			// Having more than one workloads from the same cohort admitted in the same scheduling cycle can lead
-			// to over admission if:
+			// to over admission if they have any common resources assigned and:
 			//   1. One of the workloads is borrowing, since during the nomination the usage of the other workloads
 			//      evaluated in the same cycle is not taken into account.
 			//   2. An already admitted workload from a different cluster queue is borrowing, since all workloads
 			//      evaluated in the current cycle will compete for the resources that are not borrowed.
-			if usedCohorts.Has(cq.Cohort.Name) && (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) {
-				e.status = skipped
-				e.inadmissibleMsg = "other workloads in the cohort were prioritized"
-				continue
+			if cycleCohortsUsage.hasCommonFlavors(cq.Cohort.Name, e.assignment.Usage) && (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) {
```

Comment:
```suggestion
			if cycleCohortsUsage.hasCommonResources(cq.Cohort.Name, e.assignment.Usage) && (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) {
```

Reply:
IIUC resources is what the implementation is actually checking, right?

Reply:
resources within flavors, done